pkg/data/domain/cluster/azure.go

```diff
@@ -0,0 +1,110 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/kubectl-gs/internal/label"
+)
+
+func (s *Service) getAllAzure(ctx context.Context, namespace string) (*capiv1alpha3.ClusterList, error) {
+	var err error
+
+	options := &runtimeClient.ListOptions{
+		Namespace: namespace,
+	}
+
+	clusterList := &capiv1alpha3.ClusterList{}
+	{
+		err = s.client.K8sClient.CtrlClient().List(ctx, clusterList, options)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		} else if len(clusterList.Items) == 0 {
+			return nil, microerror.Mask(noResourcesError)
+		}
+	}
+
+	var objKey runtimeClient.ObjectKey
+	for _, cluster := range clusterList.Items {
+		// FIXME(axbarsan): Remove once the name is stored in
```

Comment:
The whole configmap/cluster annotation story will be gone once https://github.com/giantswarm/giantswarm/issues/12363 is done. After that, the configmap will not be necessary anymore, the name will be stored in an annotation by default.